### DISCUSSION TITLE
Ch.2 - 06. nodeSelector로 선택한 노드 집합에 Pod 배포시에 질문

### DISCUSSION THREAD

**Discussion by riveryoon-nsuslab** [yoonjeong-kwon/fastcampus-kubernetes#28] [Q&A] [answered]
강의 내용대로 노드에 label 을 추가하고
pod 을 10개를 생성을 하였는데,
노드 2개에 고르게 생성되지 않고 한 노드에 8개 다른 한노드에 2개 이렇게 생성되는데.
왜 그런건지 알수 있을까요?

**Comment by yoonjeong-kwon** [comment, marked as answer]
@riveryoon-nsuslab 님, 안녕하세요.
노드 셀렉터를 이용해서 어떤 노드에 배포할지 최소한의 제한을 줄 수 있으나, 어떤 노드에 파드를 얼만큼 배포할 것인가는 쿠버네티스가 어떤 알고리즘에 의해 자동으로 결정하기 때문에 노드가 2개라고 해서 5개씩 균등하게 배포된다는 보장은 없습니다.
좀 더 디테일한 노드를 선택하기 위해서 다른 방법들도 제공되고 있습니다.
아래 링크를 참고해주시면 감사하겠습니다. 😃 
https://kubernetes.io/ko/docs/concepts/scheduling-eviction/assign-pod-node/

링크 첫머리에 나오는 일부 내용입니다. `특정한 노드(들)집합에서만 동작하거나 특정한 노드 집합에서 동작하는 것을 선호하도록 파드를 제한할 수 있다. 이를 수행하는 방법에는 여러 가지가 있으며 권장되는 접근 방식은 모두 레이블 셀렉터를 사용하여 선택을 용이하게 한다. 보통은 스케줄러가 자동으로 합리적인 배치(예: 자원이 부족한 노드에 파드를 배치하지 않도록 노드 간에 파드를 분배)를 수행하기에 이러한 제약 조건을 설정할 필요는 없다. 그러나, ...`
